src/past-purchase/index.ts

```diff
@@ -0,0 +1,34 @@
+import { alias, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+
+@alias('pastPurchaseItem')
+@tag('gb-sayt-past-purchase', require('./index.html'))
+class PastPurchase {
+  $pastPurchase: Store.Autocomplete.Suggestion;
+
+  state: PastPurchase.State = {
+    onClick: (event: MouseEvent) => {
+      this.flux.store.dispatch(this.flux.actions.updatePastPurchaseQuery(this.$pastPurchase.value));
+      this.flux.emit('sayt:hide');
```

Comment:
`this.flux.actions.updatePastPurchaseQuery(this.$pastPurchase.value)`, you were using `this.actions` instead of `this.flux.actions` elsewhere. You can make them consistent

Reply:
Did you want to use event.target.value? Because you have it here in the function